src/libs/actions/User.js

```diff
@@ -53,7 +53,8 @@ function changePasswordAndNavigate(oldPassword, password) {
         .then((response) => {
             if (response.jsonCode !== 200) {
                 const error = lodashGet(response, 'message', 'Unable to change password. Please try again.');
-                Onyx.merge(ONYXKEYS.ACCOUNT, {error});
+                const errorCode = lodashGet(response, 'jsonCode', 0);
+                Onyx.merge(ONYXKEYS.ACCOUNT, {error, errorInCurrentPassword: errorCode === 401, errorToNotAllowPasswordChange: errorCode === 666});
```

Comment:
I know this is a bit against what we discussed in the issue but what if instead of `errorInCurrentPassword` and `errorToNotAllowPasswordChange` we used something like `errorType`, where `errorType` can be `current`, `new`, or `generic`. And then we could follow the convention [here](https://github.com/Expensify/App/blob/42a9518275a936673b73e6b2272a90faaa3785b8/src/libs/actions/Session/index.js#L456-L461) and do something like: 

```Javascript 
            if (errorCode.jsonCode === 401) {
                Onyx.merge(ONYXKEYS.SESSION, {errorType: 'current'});
            }
            if (errorCode.jsonCode === 666) {
                Onyx.merge(ONYXKEYS.ACCOUNT, {errorType: 'generic'});
            }
```

Reply:
Resolved.